sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -113,8 +113,9 @@ case class AddFile(path: String) extends RunnableCommand {
 
   override def run(sqlContext: SQLContext): Seq[Row] = {
     val hiveContext = sqlContext.asInstanceOf[HiveContext]
+    val recursive = sqlContext.sparkContext.getConf.getBoolean("spark.input.dir.recursive", false)
```

Comment:
Adding this session-scoped configuration is risky. If needed, we can improve the SQL syntax for supporting it.